src/main/java/io/kamax/matrix/client/AMatrixHttpClient.java

```diff
@@ -61,6 +64,9 @@
 
     public AMatrixHttpClient(MatrixClientContext context) {
         this.context = context;
+        if (!context.getToken().isPresent()) {
```

Comment:
The CS API don't require login everywhere, like fetching media or getting the list of supported protocol version. This cannot be made mandatory.